python/ray/tune/impl/utils.py

```diff
@@ -26,3 +29,13 @@ def execute_dataset(config_dict: dict):
                     config_dict[k].categories = [
                         _item.fully_executed() for _item in _list
                     ]
+
+
+def process_scaling_config(params_space: dict):
+    """Convert ``params_space["scaling_config"]`` back to a dict so that
+    it can be used to generate search space.
```

Comment:
Maybe I'm missing something obvious, but when/where does this get converted back to a `ScalingConfigDataClass` to be used by the `Trainer`??

Reply:
Tuner doesn't interact with Trainer.__init__ directly does it? It only interacts with Trainable. So why not just keep this internal interface taking in a dict based ScalingConfig? I don't think we need to convert it back to dataclass. 

In short, Trainer speaks dataclass while Trainable speak dict.

Also to one of @amogkam's last comments, we could also update Trainer.__init__ to just take in dataclass in this PR.